src/lib/mina_lib/mina_lib.ml

```diff
@@ -479,7 +479,7 @@ let create_sync_status_observer ~logger ~is_seed ~demo_mode ~net
           match online_status with
           | `Offline ->
               (* nothing to do if offline before genesis *)
-              ( if after_genesis () then
+              ( if (not is_seed) && after_genesis () then
```

Comment:
I think this check should be moved further down. If the node is a seed node, we should still attempt the libp2p_helper restart when it's offline, but we shouldn't schedule the offline shutdown. I think adding a check before the match statement on line 493 would work better.